core/app/c/[communitySlug]/forms/[formSlug]/edit/EditFormTitleButton.tsx

```diff
@@ -0,0 +1,106 @@
+"use client";
+
+import { useState } from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { Button } from "ui/button";
+import { Dialog, DialogContent, DialogFooter, DialogTitle, DialogTrigger } from "ui/dialog";
+import {
+	Form,
+	FormControl,
+	FormDescription,
+	FormField,
+	FormItem,
+	FormLabel,
+	FormMessage,
+} from "ui/form";
+import { Input } from "ui/input";
+import { toast } from "ui/use-toast";
+
+import { didSucceed, useServerAction } from "~/lib/serverActions";
+import { updateForm } from "./actions";
+
+const editFormTitleSchema = z.object({
+	name: z.string().min(1, "Form name is required"),
+});
+
+const EditFormTitleButton = ({ name, communityId }: { name: string; communityId: string }) => {
+	const [isOpen, setIsOpen] = useState(false);
+
+	const form = useForm<z.infer<typeof editFormTitleSchema>>({
+		resolver: zodResolver(editFormTitleSchema),
+		defaultValues: {
+			name,
+		},
+	});
+
+	const runUpdateFormTitle = useServerAction(updateForm);
+	const onSubmit = (data: z.infer<typeof editFormTitleSchema>) => {
+		const result = runUpdateFormTitle({ communityId, name: data.name });
+		if (didSucceed(result)) {
```

Comment:
I don't completely know why but it looks like the form is both submitting successfully and throwing the unique constraint error here.

https://github.com/user-attachments/assets/9ce55d15-8fe0-4a89-9a99-d6f9a2b87a0a

Reply:
Oh you need to await the action!

Reply:
lmao